src/ripple/app/ledger/impl/LedgerCleaner.cpp

```diff
@@ -108,6 +108,7 @@ class LedgerCleanerImp : public LedgerCleaner
             wakeup_.notify_one();
         }
         thread_.join();
+        stopped();
```

Comment:
Consider
```
    if (thread_.joinable())
        thread_.join();
```

Reply:
We're assuming that `LedgerCleaner::stop` is called exactly once. I want `stop` methods to be able to assume that. It will make the transition to RAII much easier.